<a href="https://colab.research.google.com/github/TaranSchlich/Printer-Warranty-Probability-UW-Madison-MSDIA/blob/main/GB881_Assignment_3_Schlichtmann_T.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Warranty Claim Probability Analysis for Personal Printers

By: Taran Schlichtmann

## Business Context of Notebook

This notebook models the expected volume of warranty claims in the upcoming month, as well as the frequency of clogged printer nozzles within a fixed number of claims. Customers receive a one-year warranty on parts and service, with repairs handled either in-house or outsourced. Historical data on average monthly claims and clogged nozzle probability determine the probability estimates used in this analysis.


## Probability Distribution Overview

**Poisson Distribution (Discrete):** Estimates the number of times an event is likely to occur over a fixed time period. Used here to model monthly warranty claim counts.

**Binomial Distribution (Discrete):** Calculates the probability of a specific number of 'successes' in a fixed number of trials. Used here to model how many of the warranty claims are due to clogged nozzles.

## Probability Function Definitions

**CDF (Cumulative Distribution Function):** Probability that a random variable is less than or equal to a specific value.  
**PMF (Probability Mass Function):** Probability that a discrete random variable is exactly equal to a specific value.

## Importing SciPy for Probability Distributions

In [ ]:
# Importing SciPy for probability distributions

from scipy import stats

## Poisson Distribution: Monthly Warranty Claims

In [ ]:
# Creating Poisson random variable (monthly_claims_poisson)
# Probability of (monthly_claims_poisson) amount of warranty claims in the next month
# Warranty claims monthly average has been 175

monthly_claims_poisson = stats.poisson(175)

In [ ]:
# Probability of getting 200 or fewer warranty claims
# CDF stands for cumulative distribution function which provides the cumulative probability that a random variable is less than or equal to a particular value

monthly_claims_poisson.cdf(200)
round(monthly_claims_poisson.cdf(200), 3)

np.float64(0.971)

There is a 97.1% chance to receive 200 or fewer warranty claims in the next month.

In [ ]:
# Probability of getting 200 or more warranty claims
# 1 minus CDF (cumulative distribution function) provides the cumulative probability that a random variable is more than or equal to a particular value

1 - monthly_claims_poisson.cdf(200)
round(1 - monthly_claims_poisson.cdf(200), 3)

np.float64(0.029)

There is a 2.9% chance to receive 200 or more warranty claims.

In [ ]:
# Probability of getting exactly 175 warranty claims
# PMF stands for probability mass function which is used to describe the distribution of a discrete random variable

monthly_claims_poisson.pmf(175)
round(monthly_claims_poisson.pmf(175), 3)

np.float64(0.03)

There is a 3% chance to receive exactly 175 warranty claims.

## Binomial Distribution: Clogged Nozzle Claims

In [ ]:
# Clogged nozzles account for 25% of claims
# Creating a binomial random variable (clogged_nozzle_binomial)
# Looking at the probability over the next 300 warranty claims

clogged_nozzle_binomial = stats.binom(300, 0.25)

In [ ]:
# Probability exactly 70 of the next 300 warranty claims related to clogged printer nozzles

clogged_nozzle_binomial.pmf(70)
round(clogged_nozzle_binomial.pmf(70), 3)

np.float64(0.043)

There is a 4.3% chance exactly 70 of the next 300 warranty claims will be related to clogged printer nozzles.

In [ ]:
# Probability more than 80 of the next 300 warranty claims related to clogged printer nozzles

1 - clogged_nozzle_binomial.cdf(80)
round(1 - clogged_nozzle_binomial.cdf(80), 3)

np.float64(0.23)

There is a 23% chance more than 80 of the next 300 warranty claims will be related to clogged printer nozzles.

In [ ]:
# Probability between 75 and 90 of the next 300 warranty claims related to clogged printer nozzles (include 75 and 90)
# Note the use of 74 instead of 75 in the calculation to include 75 in our calculation

clogged_nozzle_binomial.cdf(90) - clogged_nozzle_binomial.cdf(74)
round((clogged_nozzle_binomial.cdf(90) - clogged_nozzle_binomial.cdf(74)), 3)

np.float64(0.501)

There is a 50.1% chance between 75 and 90 of the next 300 warranty claims will be related to clogged printer nozzles.

## Summary

This notebook applied discrete probability models to estimate monthly warranty claim volumes and the frequency of clogged printer nozzle incidents. Using Poisson and Binomial distributions, the analysis shows that receiving 200 or fewer claims in a month is highly probable, while clogged nozzles are expected to account for roughly one-quarter of all claims. These insights support data-driven planning for service operations based on the historical data provided.